### ISSUE TITLE
Validator YD01 behavior on yields None

### ISSUE THREAD

**Issue by jonyscathe** [numpy/numpydoc#580]
In numpydoc-validator if I have something like:

```
@pytest.fixture
def db_setup() -> Generator[None, None, None]:
    """
    Add test user.

    Clean up test used at the end of the test.
    """
    database = Database('my_database_url')
    with database.create_session() as session:
        session.execute(insert(database.Users, [{'id': 1, 'name': 'Bob'}

    yield

    with database.create_session() as session:
        session.execute(delete(database.Users).where(database.Users.id == 1234)
```
Then I get:
```
+---------------------+-------------------+---------+-------------------------+
| file                | item              | check   | description             |
+=====================+===================+=========+=========================+
| src/conftest.py:135 | conftest.db_setup | YD01    | No Yields section found |
+---------------------+-------------------+---------+-------------------------+
```
Is this expected behavior?
From the numpdoc [Yields](https://numpydoc.readthedocs.io/en/latest/format.html#yields) section, the Yields section is meant to explain the yielded values and their types.
But in this case there are no yielded values or types, the yield is just used to split up the test setup from the test teardown in a pytest fixture.

I would have thought that this was similar to a function with return type of None and in that case the numpydoc-validator tool does not expect an empty returns section.

**Comment by larsoner** [comment]
Well at least the undecorated function yields a single element which is None, e.g. to simplify the use case and look at it differently:
```
>>> def yielder():
...     yield
... 
>>> list(yielder())
[None]
```
But then pytest does some magic that numpydoc might not be able to make sense of. I think this is one of those cases where you probably want to add an [inline ignore](https://numpydoc.readthedocs.io/en/latest/validation.html#inline-ignore-comments), and there isn't much to change at the numpydoc end.

**Comment by jonyscathe** [comment]
I don't think pytest has anything to do with this.

```
def yielder() -> Generator[None, None, None]:
    """Test."""
    yield
```
Still results in:
```
+---------------------+-------------------+---------+-------------------------+
| file                | item              | check   | description             |
+=====================+===================+=========+=========================+
| src/conftest.py:135 | conftest.yielder  | YD01    | No Yields section found |
+---------------------+-------------------+---------+-------------------------+
```

Is this expected behaviour? It isn't yielding anything so why is a yields section required in the docstring.

**Comment by stefanv** [comment]
How do you recommend we handle these? It needs to be documented somewhere that the function is a generator.

**Comment by larsoner** [comment]
> How do you recommend we handle these? It needs to be documented somewhere that the function is a generator.

Agreed, I was just about to say...

> Is this expected behaviour? It isn't yielding anything so why is a yields section required in the docstring.

... it is implicitly yielding one `None`. That's what I was trying to suggest with my example. To make it a bit clearer perhaps, this version yields 10 implicit `None`s via a generator:
```
>>> def yielder():
...     for _ in range(10):
...         yield
...
>>> list(yielder())
[None, None, None, None, None, None, None, None, None, None]
```
It seems reasonable that this and your example would both need some `Yields` section of some sort to tell people about the nature of the returned generator.

**Comment by jonyscathe** [comment]
@stefanv in this situation what should be in the yields section of the docstring.

The example in the nymodoc style guide is:
```
Yields
------
int
    Description of the anonymous integer return value.
```

The equivalent here would be:
```
Yields
------
None
    Yields nothing.
```
Which is actually about what we had.
Main reason that I raised this was because ruff started throwing DOC403 on those docstrings.
I raised an issue there first (https://github.com/astral-sh/ruff/issues/13001) but they stated that they did this on purpose, much the same as they don't expect to have a Returns section in a docstring if the function returns None.

I guess we'll just have to chose which rule to ignore between DOC403 in ruff or YD01 here.
